### DISCUSSION TITLE
Help with copy files on MacOS

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3784] [Extension Q&A] [answered]
I'm trying to Move  a file and rename it at the same time. This is the Provision Profile folder which is hidden by default.

`		string provPath = $"{directory.ToString()}/{BranchProject}/adhoc.mobileprovision";
		//"mkdir -p ~/Library/MobileDevice/Provisioning\\ Profiles".Bash();
		$"security cms -D -i \"{provPath.ToString()}\"".Bash();

		string uuid = $"security cms -D -i \"{provPath}\" | grep -aA1 UUID | grep -o \"[-a-zA-Z0-9]\\{{36\\}}\"".Bash();
		Console.WriteLine($"UUID found: {uuid}");

		// string command = $"cp \"{provPath}\" \"~/Library/MobileDevice/Provisioning\\ Profiles/{uuid}.mobileprovision\"";
		Console.WriteLine($"copy .{provPath} ~/Library/MobileDevice/Provisioning Profiles/{uuid}.mobileprovision");

		MoveFile($"{provPath}", $"~/Library/MobileDevice/Provisioning Profiles/{uuid}.mobileprovision");`

But when I execute the move file I get this error:
An error occurred when executing task 'InstallIOSProvision'.
Error: System.AggregateException: One or more errors occurred. (The directory '/Library/agents/_work/7/s/build/~/Library/MobileDevice/Provisioning Profiles' does not exist.) ---> System.IO.DirectoryNotFoundException: The directory '/Library/agents/_work/7/s/build/~/Library/MobileDevice/Provisioning Profiles' does not exist.

How can I specify the root folder on the MacOS?

**Comment by augustoproiete** [comment]
An easy way is to leverage the [`Cake.PathTildeHelper`](https://github.com/cake-contrib/Cake.PathTildeHelper) addin which exposes the alias `PathReplaceTilde` for resolving the user profile path as needed:

```cakescript
#addin "nuget:?package=Cake.PathTildeHelper&version=4.0.0"

Information(PathReplaceTilde(new FilePath("~/Documents/awesome-stuff.txt")));
```

Alternatively, if you want to resolve it yourself using .NET/C#, you can call [`Environment.GetFolderPath`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getfolderpath?view=net-6.0):

```csharp
var userProfileFolder = System.Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile);
```

**Reply by nick5454** [reply]
@augustoproiete I'm trying to get to the root folder of the MacOs main drive, which is where the folder resides for all users. This is before the uSers tree in the hierarchy

**Comment by augustoproiete** [comment, marked as answer]
> I'm trying to get to the root folder of the MacOs main drive, which is where the folder resides for all users. This is before the uSers tree in the hierarchy

Gotcha. I saw you using `~` which would resolve to the user profile folder and thought you were having issues with Cake doing that.

In MacOS the root folder would be `/System`:

```cakescript
var filePath = new FilePath("/System/Library/Fonts/Supplemental/Arial Unicode.ttf");
CopyFile(filePath, "./MyFont.ttf");
```
